File: 백준/Silver/10773. 제로/제로.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.Stack;

public class Main {
    Stack<Integer> stack = new Stack<>();

    public static void main(String[] argus) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int firstV = Integer.parseInt(br.readLine());

        Main main = new Main();
        long outPut = 0;
        for (int i = 0; i < firstV; i++) {
            int a = Integer.parseInt(br.readLine());
            outPut += main.solution(a);
        }
        System.out.println(outPut);
    }

    private long solution(int input) {
        long sum = 0;
        if (input != 0) {
            stack.push(input);
            sum += input;
        } else {
            Integer pop = stack.pop();
            sum -= pop;
        }
        return sum;
    }
}

File: 백준/Silver/10773. 제로/제로.py
# 제로
import sys
def input():
    return sys.stdin.readline().rstrip()


answer_list=[0]

N=int(input())

for _ in range(N):
    number = int(input())
    if number ==0:
        answer_list.pop()
    else:answer_list.append(number)


print(sum(answer_list))